presto-main/src/test/java/com/facebook/presto/cost/TestScalarStatsCalculator.java

```diff
@@ -316,6 +316,39 @@ public void testNonDivideArithmeticBinaryExpression()
                 .averageRowSize(2.0);
     }
 
+    @Test
+    public void testArithmeticBinaryWithZero()
+    {
+        PlanNodeStatsEstimate relationStats = PlanNodeStatsEstimate.builder()
+                .addSymbolStatistics(new Symbol("x"), SymbolStatsEstimate.builder()
+                        .setLowValue(-1)
+                        .setHighValue(10)
+                        .setDistinctValuesCount(4)
+                        .setNullsFraction(0.1)
+                        .setAverageRowSize(0)
+                        .build())
+                .addSymbolStatistics(new Symbol("zero"), SymbolStatsEstimate.zero())
+                .setOutputRowCount(10)
+                .build();
+
+        assertCalculate(expression("x + zero"), relationStats)
+                .isEqualTo(SymbolStatsEstimate.zero());
+        assertCalculate(expression("x - zero"), relationStats)
+                .isEqualTo(SymbolStatsEstimate.zero());
+        assertCalculate(expression("zero - x"), relationStats)
+                .isEqualTo(SymbolStatsEstimate.zero());
+        assertCalculate(expression("zero * x"), relationStats)
+                .isEqualTo(SymbolStatsEstimate.zero());
+        assertCalculate(expression("x % zero"), relationStats)
+                .isEqualTo(SymbolStatsEstimate.zero());
+        assertCalculate(expression("zero % x"), relationStats)
+                .isEqualTo(SymbolStatsEstimate.zero());
+        assertCalculate(expression("x / zero"), relationStats)
+                .isEqualTo(SymbolStatsEstimate.zero());
```

Comment:
this should be `unknowns` or `NaN`

Reply:
Any arithmetic operation with `all_nulls` column will result in `all_nulls` results. Let me do the renames.